src/Skinner.js

```diff
@@ -42,7 +48,25 @@ class Skinner {
                 "If you want to change the active skin, call resetSkin first"
             );
         }
-        this.components = skinObject;
+        this.components = {};
+        var compKeys = Object.keys(skinObject.components);
+        for (var i = 0; i < compKeys.length; ++i) {
+            var comp = skinObject.components[compKeys[i]];
+            this.addComponent(compKeys[i], comp);
+        }
+    }
+
+    addComponent(name, comp) {
+        var slot = name;
+        if (comp.replaces !== undefined) {
```

Comment:
I'm not sure I like this. AFAIK this is so we can declare that component X clobbers component Y programatically so the reskindex process can dump the right things in place. This is subtle and makes it hard to see what is going on. This is exacerbated by `:65` which by the looks of it does a last path segment match if there is no `.`.

We presumably need this because vector-web's `RoomDNDView` clobbers `rooms.RoomTile`. I still feel that this is best represented as simply:
```
module.exports.components['views.rooms.RoomTile'] = require('./components/views/rooms/RoomDNDView');
```
It's clear, familiar and unambiguous. Does the *entire* `component-index.js` have to be auto-genned like this? I'd much prefer a separate file e.g. `component-overwrites.js` which allows you to assign these kinds of clobbers without needing to know the magic `.replaces` property, special-case last path segment handling, etc.